chunk/by_key_test.go

```diff
@@ -9,7 +9,70 @@ func c(id string) Chunk {
 	return Chunk{UserID: id}
 }
 
-func TestIntersect(t *testing.T) {
+func TestUnique(t *testing.T) {
+	for _, tc := range []struct {
+		in   ByKey
+		want ByKey
+	}{
+		{nil, ByKey{}},
+		{ByKey{c("a"), c("a")}, ByKey{c("a")}},
+		{ByKey{c("a"), c("a"), c("b"), c("b"), c("c")}, ByKey{c("a"), c("b"), c("c")}},
+		{ByKey{c("a"), c("b"), c("c")}, ByKey{c("a"), c("b"), c("c")}},
+	} {
+		have := unique(tc.in)
+		if !reflect.DeepEqual(have, tc.want) {
+			t.Errorf("%v != %v", have, tc.want)
+		}
+	}
+}
+
+func TestMerge(t *testing.T) {
+	type args struct {
+		a ByKey
+		b ByKey
+	}
+	for _, tc := range []struct {
+		args args
+		want ByKey
+	}{
+		{args{ByKey{}, ByKey{}}, ByKey{}},
+		{args{ByKey{c("a")}, ByKey{}}, ByKey{c("a")}},
+		{args{ByKey{}, ByKey{c("b")}}, ByKey{c("b")}},
+		{args{ByKey{c("a")}, ByKey{c("b")}}, ByKey{c("a"), c("b")}},
+		{
+			args{ByKey{c("a"), c("c")}, ByKey{c("a"), c("b"), c("d")}},
+			ByKey{c("a"), c("b"), c("c"), c("d")}},
+	} {
+		have := merge(tc.args.a, tc.args.b)
+		if !reflect.DeepEqual(have, tc.want) {
+			t.Errorf("%v != %v", have, tc.want)
+		}
+	}
+}
```

Comment:
This is a really minor point, but your `nWayMerge` algorithm assumes `ByKey` and `merge` form a monoid. You should probably have a test for associativity and identity properties. (I think the code works and the test coverage is enough, so no need to actually _do_ this).